### ISSUE TITLE
[Android] Java.Lang.RuntimeException: Cursor window could not be created from binder.

### ISSUE THREAD

**Issue by alexandermontgomery** [SimonSimCity/Xamarin-CrossDownloadManager#49]
Hey All - Love the library. It is good stuff. I am having one issue though. I am seeing this crash report in my app:

```
Xamarin caused by: Java.Lang.RuntimeException: Cursor window could not be created from binder.
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()<eff4b620bf524ecf9adcac0016ee2f1e>:0
Java.Interop.JniEnvironment.InstanceMethods.CallObjectMethod(JniObjectReference instance, JniMethodInfo method, JniArgumentValue* args)<ee12d51b0c0e4eba9f8e647503740099>:0
Java.Interop.JniPeerMembers.JniInstanceMethods.InvokeVirtualObjectMethod(string encodedMember, IJavaPeerable self, JniArgumentValue* parameters)<ee12d51b0c0e4eba9f8e647503740099>:0
Android.App.DownloadManager.InvokeQuery(DownloadManager.Query query)<0acc886950d94aac8d55fa6bbaa8e8a2>:0
Plugin.DownloadManager.DownloadManagerImplementation.LoopOnDownloads(Action<T> runnable)<30ccf818685c43d093a28f9c7a1e9397>:0
Plugin.DownloadManager.DownloadManagerImplementation.<StartDownloadWatcher>m__1()<30ccf818685c43d093a28f9c7a1e9397>:0
Java.Lang.Runnable.Run()<0acc886950d94aac8d55fa6bbaa8e8a2>:0
Java.Lang.IRunnableInvoker.n_Run(IntPtr jnienv, IntPtr native__this)<0acc886950d94aac8d55fa6bbaa8e8a2>:0
```

Digging a bit more into your library I can see in this file https://github.com/SimonSimCity/Xamarin-CrossDownloadManager/blob/03864003db7d5034786fa27eafe415572791755c/DownloadManager/Plugin.DownloadManager.Android/DownloadManagerImplementation.cs the crash is happening somewhere inside of StartDownloadWatcher.

All I have found related to this is this Stackoverflow question: http://stackoverflow.com/questions/18502578/android-app-crashing-in-cursor-query-image-video-cursor

I thought I would reach out to you guys to see if you have any insight yourself.

This happened on a Samsung Galaxy S8 running Android 7.

I have not been able to reproduce this on the same device and OS.

Thoughts?

**Comment by SimonSimCity** [comment]
The stackoverflow you've linked here could well be connected to this issue.

The method `StartDownloadWatcher()` calls the method `LoopOnDownloads()`, where it starts to query the download-manager in line 99 https://github.com/SimonSimCity/Xamarin-CrossDownloadManager/blob/03864003db7d5034786fa27eafe415572791755c/DownloadManager/Plugin.DownloadManager.Android/DownloadManagerImplementation.cs#L99.

If it is a problem with memory, you should make sure you have a lot of downloads queued before crashing and opening your app. I have to take a deeper look at what the query actually is running over. Maybe you'll be able to reproduce it then ... Give me some time on that.

I've actually seen this one a lot on various devices but was not even able to get a good stack to confirm where the error came from.

**Comment by alexandermontgomery** [comment]
Simon - thanks for getting back to me! I will continue to mess around with
it and see what it could be. The 'lots of files queued' is the obvious
answer but to run out of memory on a query like that it either means:

1) each file result from query includes the actual file binary in the
result object
2) There are 500+ files queued

I will try to look more into it but any help from you is much appreciated.

**Comment by cavaIV** [comment]
This error still occurs
Any updates?

**Comment by SimonSimCity** [comment]
This bug is related on some code that I refactored lately while trying to fix #77 (see c12a0f1a7dcb8e291a6c6436bc9660f834e7b25e).

@cavaIV If you have an environment where this exception occurs quite frequently, I ask you to test it with the code currently on the `develop` branch of this repo.

If you just found this in the exception-reporter (e.g. Xamarin Insights) I kindly ask you to wait for my release until I have confirmed that it at least solves the issue #77.
